api/restHandler/app/appList/AppListingRestHandler.go

```diff
@@ -990,13 +992,28 @@ func (handler AppListingRestHandlerImpl) getAppDetails(ctx context.Context, appI
 	return appDetail, err, appId
 }
 
+func getApplicationStatusFromLatestDeployment(latestDeployment *chartConfig.LatestDeployment) string {
+	if latestDeployment != nil {
+		if latestDeployment.RunnerStatus == argoApplication.Progressing {
```

Comment:
if the last non failed(definition according to db method name) is in progressing we cannot ensure it went ahead for deployment, need to check the actual latest runner to dedupe and timeline